packages/sdks/web-component/src/lib/helpers/templates.ts

```diff
@@ -47,11 +47,24 @@ const replaceElementInputs = (
 /**
  * Get object nested path.
  * Examples:
- *  - getByPath({ { a { b: 'rob' } }, 'a.b') => 'hey rob'
+ *  - getByPath({ { a { b: 'rob' } }, 'a.b') => 'rob'
+ *  - getByPath({ { a { 'b.c': 'rob' } }, 'a.b.c') => 'rob'
  *  - getByPath({}, 'a.b') => ''
  */
 const getByPath = (obj: Record<string, any>, path: string) =>
-  path.split('.').reduce((prev, next) => prev?.[next] || '', obj);
+  {
```

Comment:
Why do we had to change it?
For which cases it wouldn't work?

Reply:
cases where the key may contain `.` or the screen state is flatten for example `{ "form": { "test.id": "value" } }`. seems like the screen values support that so i preferred to change the `getByPath` to support that as well

Reply:
Thanks for the clarification

I think it won't work for some objects,
e.g.
```
{a: {'b.c': {d: 1}}}

{'a.b': {'c.d': 1}}
```
Do we want to support it?

In general, I think there is a problem here, 
if the keys might contains dots, there are many possibilities for each received path, and I don't think we are covering them all

WDYT?

Reply:
you are right, although this is not possible because the screen state must contain the key first (such as form / client etc) but this handle internal screen state that is generate by the sdk so this internal logic can come from multiple places

Reply:
Not sure I understand
Do you think it's problematic?